src/main/java/io/quarkus/registry/app/model/PlatformRelease.java

```diff
@@ -137,17 +147,42 @@ public static Optional<PlatformRelease> findByNaturalKey(PlatformStream platform
     }
 
     public static List<PlatformRelease> findLatest(String quarkusCore) {
-        EntityManager entityManager = getEntityManager();
+        EntityManager em = getEntityManager();
         TypedQuery<PlatformRelease> query;
-        if (quarkusCore != null && !quarkusCore.isBlank()) {
-            query = entityManager.createNamedQuery("PlatformRelease.findLatestByQuarkusCoreVersion", PlatformRelease.class)
+        boolean withQuarkusCore = quarkusCore != null && !quarkusCore.isBlank();
+        if (withQuarkusCore) {
+            query = em.createNamedQuery("PlatformRelease.findLatestByQuarkusCoreVersion", PlatformRelease.class)
                     .setParameter(1, quarkusCore);
         } else {
-            query = entityManager.createNamedQuery("PlatformRelease.findLatest", PlatformRelease.class);
+            query = em.createNamedQuery("PlatformRelease.findLatest", PlatformRelease.class);
+        }
+        // Check if we can perform this directly in the SQL
+        // We need the top 3 releases for each platform
+        var platformMap = new LinkedHashMap<Platform, List<PlatformRelease>>();
+        for (var item : query.getResultList()) {
+            var platformReleases = platformMap.computeIfAbsent(item.platformStream.platform,
+                    k -> new ArrayList<>());
+            if (platformReleases.size() < 3) {
+                // Count stable versions
+                long stableVersions = platformReleases.stream()
+                        .filter(pr -> !Version.isPreFinal(pr.version))
+                        .count();
+                if (stableVersions < 2) {
+                    platformReleases.add(item);
```

Comment:
It hasn't happened before but in case there are more than 2 non-final items in a raw (e.g. CR1, CR2) will we end up with a single stable release?

Reply:
The SQL will only return the latest and greatest (in this example, CR2)